management_app/views/course.py

```diff
@@ -2,22 +2,140 @@
 
 courses = Blueprint('courses', __name__, url_prefix='/courses')
 
-@courses.route('/')
+@courses.route('/offerings', methods=['GET'])
 @login_required
-def index():
-    db = get_db()
-    # TODO: should filter by the year selected from the frontend
-    courses = db.execute(
-            'SELECT year, quarter, user_name, course_title_id, course_sec, enrollment'
-            ' FROM scheduled_teaching st JOIN courses ON st.course_id = courses.course_id JOIN users ON st.user_id = users.user_id'
-            ' WHERE year = 2022'#, (,)
-    ).fetchall()
+def offerings():
+    if request.method == 'GET':
+        # generate year options
+        todays_date = date.today()  # creating the date object of today's date
+        years = list(range(2019, todays_date.year+1))  # [2019, 2020, 2021, 2022]
+        year_options = [str(y) + '-' + str(y+1) for y in years]  # ['2019-2020', '2020-2021', '2021-2022', '2022-2023']
 
-    return render_template('courses/index.html', courses=courses)
+        db = get_db() 
+        y_start, y_end = todays_date.year, todays_date.year+1
+        year_selected = request.args.get('year')  # 2020-2021 (including: 2020 Fall - 2021 Winter & Spring)
+        if year_selected != None:
+            y_start = year_selected.split('-')[0]
+            y_end = year_selected.split('-')[1]
 
+        courses = db.execute(
+                'SELECT year, quarter, user_name, course_title_id, course_sec, enrollment'
+                ' FROM scheduled_teaching st JOIN courses ON st.course_id = courses.course_id JOIN users ON st.user_id = users.user_id'
+                ' WHERE (year = ? AND quarter = 1) OR (year = ? AND quarter = 2) OR (year = ? AND quarter = 3)', (y_start, y_end, y_end)
+        ).fetchall()
+
+        return render_template('courses/offerings.html', courses=courses, year_options=year_options)
+
+
+@courses.route('/catalog', methods=['GET'])
+@login_required
+def catalog():
+    if request.method == 'GET':
+        # # Initial courses table: parse input file and insert data into db only for the first time
+        # db = get_db()
+
+        # df = pd.read_excel(get_upload_filepath("courses.xlsx"), sheet_name=1)
+        # rows = df.values.tolist()   
+        # for row in rows:
+        #     # course_title_id column may be like "CS 143B" or "CS143B" -> all convert to "CS143B"
+        #     course_title_id = row[0].strip().replace(' ', '')
+
+        #     course_title = row[1].strip()
+        #     units = row[2]            
+        #     course_level = row[3].strip()
+        #     teaching_point_val = -1
+
+        #     db.execute(
+        #         'INSERT INTO courses (course_title_id, course_title, units, teaching_point_val, course_level)'
+        #         ' VALUES (?, ?, ?, ?, ?)',
+        #         (course_title_id, course_title, units, teaching_point_val, course_level)
+        #     )
+        #     db.commit()
+
+        # TODO: haven't calculate "teaching_point_val" column
```

Comment:
Suggestion but minor: same here for the comment. Those codes in comments can push in the next commit, so it could be more readable.

You may want to commit to course catalog features next time, so let me write the relevant question here. I was wondering how will we store "teaching_point_val" in the database course table.

We probably need to give the default `teaching_point_val = 1` for each course before week 2. Only when there is an update for actual enrollment by the "Schedule_teaching" template, this value will be automatically updated by the "get_course_point_tier" function (once we have it). Let's discuss this offline for more details.